From b109f0c78f46a2d54af529ceb71c69a1476b24ec Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Fri, 7 Mar 2008 22:54:32 +0100
Subject: [PATCH] makepkg: move strict-aliasing hack here
 * while we still need -fno-strict-aliasing
 * moved the hack from util.sh ( where it never worked )
 * to here. We can't do that from util.sh because
 * makepkg.conf is sourced after util.sh so the *FLAGS are lost

---
 scripts/makepkg |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 5333002..a84c3ab 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -114,6 +114,14 @@ Fextract() {
 # makepkg configuration
 [ -f /etc/makepkg.conf ] && source /etc/makepkg.conf
 
+# we need -fno-strict-aliasing for gcc 4.2*++
+_gccver=`type -p gcc >/dev/null && gcc -dumpversion`
+if [ "${_gccver%%.?}" == "4.3" ]; then
+        export CFLAGS="$CFLAGS -fno-strict-aliasing"
+        export CXXFLAGS="$CXXFLAGS -fno-strict-aliasing"
+fi
+
+
 INFAKEROOT=
 if [ "$1" = "-F" ]; then
 	INFAKEROOT=1
-- 
1.5.4.3

